tutorials/tts/2_Inference_DurationPitchControl.ipynb

```diff
@@ -0,0 +1,490 @@
+    "\n",
+    "# Let's run fastpitch normally\n",
+    "_, audio, *_ = str_to_audio(input_string)\n",
+    "print(f\"This is fastpitch speaking at the regular pace of 1.0. This example is {len(audio[0])/sr:.3f} seconds long.\")\n",
+    "ipd.display(ipd.Audio(audio, rate=sr))\n",
+    "\n",
+    "# We can speed up the speech by adjusting the pace\n",
+    "_, audio, *_ = str_to_audio(input_string, pace=1.2)\n",
+    "print(f\"This is fastpitch speaking at the faster pace of 1.2. This example is {len(audio[0])/sr:.3f} seconds long.\")\n",
+    "ipd.display(ipd.Audio(audio, rate=sr))\n",
+    "\n",
+    "# We can slow down the speech  by adjusting the pace\n",
+    "_, audio, *_ = str_to_audio(input_string, pace=0.75)\n",
+    "print(f\"This is fastpitch speaking at the faster pace of 0.75. This example is {len(audio[0])/sr:.3f} seconds long.\")\n",
+    "ipd.display(ipd.Audio(audio, rate=sr))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pitch Control\n",
+    "\n",
+    "The newer spectrogram generator models predict the pitch for certain words. Since these models predict pitch, we can adjust the predicted pitch in a similar manner to the predicted durations like in the previous section. A list of NeMo models that support pitch control are as follows:\n",
+    "\n",
+    "- [FastPitch](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/models/fastpitch.py)\n",
+    "- [FastPitch_HifiGan_E2E](https://ngc.nvidia.com/catalog/models/nvidia:nemo:tts_en_e2e_fastpitchhifigan)\n",
+    "- [FastSpeech2](https://ngc.nvidia.com/catalog/models/nvidia:nemo:tts_en_fastspeech_2)\n",
+    "- [FastSpeech2_HifiGan_E2E](https://ngc.nvidia.com/catalog/models/nvidia:nemo:tts_en_e2e_fastspeech2hifigan)\n",
+    "- [TalkNet](https://ngc.nvidia.com/catalog/models/nvidia:nemo:tts_en_talknet)\n",
+    "\n",
+    "### FastPitch\n",
+    "\n",
+    "As with the previous tutorial, we will focus on FastPitch. FastPitch differs from some other models as it predicts a pitch difference to a normalized (mean 0, std 1) speaker pitch. Other models will just predict the unnormalized pitch. Looking at a simplified version of the FastPitch model, we see\n",
+    "\n",
+    "```python\n",
+    "# Predict pitch\n",
+    "pitch_predicted = self.pitch_predictor(enc_out, enc_mask)  # Predicts a pitch that is normalized with speaker statistics \n",
+    "pitch_emb = self.pitch_emb(pitch.unsqueeze(1))  # A simple 1D convolution to map the float pitch to a embedding pitch\n",
+    "\n",
+    "enc_out = enc_out + pitch_emb.transpose(1, 2)  # We add the pitch to the encoder output\n",
+    "spec, *_ = self.decoder(input=len_regulated, seq_lens=dec_lens)  # We send the sum to the decoder to get the spectrogram\n",
+    "```\n",
+    "\n",
+    "Let's see the `pitch_predicted` for a sample text. You can run the below cell. You should get an image that looks like for the input `Hey, what is my pitch?`:\n",
```

Comment:
Typo?
s/looks like for/looks like the following for

Reply:
done